metadata_input_netlink.c

```diff
@@ -109,6 +106,18 @@ static uint8_t md_input_netlink_parse_conn_event(struct md_input_netlink *min,
 
         if (!strcmp(key, "tx_bytes"))
             mce->tx_bytes = (uint64_t) json_object_get_int64(val);
+
+        if (!strcmp(key, "has_ip"))
+            mce->has_ip = (uint8_t) json_object_get_int(val);
+
+        if (!strcmp(key, "connectivity"))
+            mce->connectivity = (uint8_t) json_object_get_int(val);
+
+        if (!strcmp(key, "connection_mode"))
+            mce->connection_mode = (uint8_t) json_object_get_int(val);
+
+        if (!strcmp(key, "quality"))
+            mce->quality = (uint8_t) json_object_get_int(val);
```

Comment:
We should rewrite this to be an else-if. This applies to all the ifs inside the loop and it is my mistake :)